esmf_regrid/experimental/io.py

```diff
@@ -158,17 +258,14 @@ def _standard_grid_cube(grid, name):
         long_name=WEIGHTS_SHAPE_NAME,
     )
 
-    # Avoid saving bug by placing the mesh cube second.
-    # TODO: simplify this when this bug is fixed in iris.
-    if regridder_type == "GridToMeshESMFRegridder":
+    # Save cubes while ensuring var_names do not conflict for the sake of consistency.
+    with _managed_var_name(src_cube, tgt_cube):
         cube_list = CubeList([src_cube, tgt_cube, weights_cube, weight_shape_cube])
-    elif regridder_type == "MeshToGridESMFRegridder":
-        cube_list = CubeList([tgt_cube, src_cube, weights_cube, weight_shape_cube])
 
-    for cube in cube_list:
-        cube.attributes = attributes
+        for cube in cube_list:
+            cube.attributes = attributes
```

Comment:
I presume that the purpose of assigning the same attributes to all cubes is to ensure that they are all made global (file) attributes?
It would be nice to leverage the new global+local attributes management to make that clearer here,
E.G. I would suggest you could just 
`src_cube.attributes = CubeAttrsDict(globals=attributes)`

However, possibly we can't do this, if we intend to still support Iris versions <3.8 ??

Reply:
I believe that is the idea, this should put attributes in a more readable place in the file (and keep the structure more similar to ESMFs file structure). I think that would be a good idea for a future change, but is probably out of scope for this PR since it does change the dependencies.